data/mods/gen2/statuses.js

```diff
@@ -12,10 +12,16 @@ let BattleStatuses = {
 		},
 		onAfterMoveSelfPriority: 3,
 		onAfterMoveSelf(pokemon) {
-			this.damage(pokemon.maxhp / 8);
+			let toxicCounter = 1;
+			if (pokemon.volatiles['residualdmg']) {
+				pokemon.volatiles['residualdmg'].counter++;
+				toxicCounter = pokemon.volatiles['residualdmg'].counter;
+			}
+			this.damage(this.clampIntRange(Math.floor(pokemon.maxhp / 8), 1) * toxicCounter, pokemon);
```

Comment:
Copied this logic from gen1, though I had some questions

- it doesn't cap the counter, but I assume this doesn't matter (below, the `effectData.stage` was capped to 15)
- it clamps the range - I think this is desirable
- I've added a hint after @Zarel asked. Should Gen 1 do this as well, or is Toxic's glitchiness well enough understood in that gen that its redundant? I purposefully placed the `-hint` so that it occurs after the `damage` message in the log, as opposed to issuing it above when we already know `toxicCounter > `.
- is it correct that the counter is only increased/applied in `onAfterMoveSelf` and not `onAfterSwitchSelf` in both gens?

Reply:
I'd like a hint in Gen 1, too. PS generally needs more hints.

Reply:
> 
> 
> Copied this logic from gen1, though I had some questions
> 
>     * it doesn't cap the counter, but I assume this doesn't matter

This is correct; the Toxic counter only started being capped to 15 in Gen 3.
>     * is it correct that the counter is only increased/applied in `onAfterMoveSelf` and not `onAfterSwitchSelf` in both gens?

If a previously burned/poisoned Pokemon is switched in with Baton Pass from a Pokemon with a Toxic counter, it immediately takes `1/16 * X` damage (even though regular poison and burn is 1/8 max HP in Gen 2). If they get passed a Toxic counter and get burned or poisoned later, it's also `1/16 * X`, where X is whatever the previous counter was + 1 no matter how many turns have passed.

Unless I'm misunderstanding, `onAfterSwitchSelf` isn't relevant in Gen 1 because Baton Pass doesn't exist.